building-and-packaging/material/packaging_python_slides.md

```diff
@@ -0,0 +1,294 @@
+- A package installed by `setup.py` needs to be manually maintained and uninstalled if required
+- Requires manual downloading of files from the package website
+
+<span>
+Is there a better way?
+<!-- .element: class="fragment" data-fragment-index="1" --></span>
+
+<span>
+Yes! Use pip!
+<!-- .element: class="fragment" data-fragment-index="2" --></span>
+
+---
+
+## What is pip?
+
+- pip is a package installer to install packages from the [Python Package Index PyPI]((https://pypi.org/))
+- pip is itself a [package](https://pypi.org/project/pip/) which is available on PyPI
+- pip is open-source and is developed on [GitHub](https://github.com/pypa/pip)
+
+---
+
+## Using pip
+
+- When `pip install package-name` is run, pip chooses a distribution file and installs it in the environment
+- Doing `python -m pip install package-name` is basically the same as `pip install package-name`
+- pip tracks metadata to allow for easy uninstallation and updating of packages
+- pip is bundled together with Python 3.x, making it even more easier to use
+- pip can install a package from a source distribution (`.tar.gz`) or a wheel distribution (`.whl`)
+- Do not use `sudo pip install`! Various security issues with doing so. Go for `pip install --user <package-name>`
+
+---
+
+## What is PyPI?
+
+- [PyPI](https://pypi.org/) = **The Python Package Index** is a repository of software developed in the Python community
+- PyPI itself is developed on GitHub through another software called [Warehouse](https://github.com/pypa/warehouse)
+- PyPI has a informative [public dashboard](https://p.datadoghq.com/sb/7dc8b3250-85dcf667bd) to show its activity
+- A major advantage is the active maintenance of PyPI and the packages indexed in it
+- Not to be confused with **PyPA** which is Python Packaging Authority, a working group which maintains projects in Python packaging
+
+---
+
+## Example of PyPI package: fenicsprecice
+
+- Having a look at [fenicsprecice](https://pypi.org/project/fenicsprecice/)
+
+---
+
+## Python Enhancement Proposals (PEPs)
+
+- PEP is an evolving design document which provides information regarding new features of Python, new processes and new environments
+- PEPs typically involve concise technical information, which also acts as standardizations
+- Packaging workflows are also standardized through PEPs. Example:
+    - [PEP 427](https://www.python.org/dev/peps/pep-0427/) introduces the built-package format "wheel"
+- It is always a good idea to look for **PEP** conventions before trying out processes
+
+---
+
+## File structure for code packaging 1/2
+
+- Most essential component of packaging is file structure. Basic file structure is as follows:
+
+```bash
+generic_folder_name/
+└── src/
+    └── package_name/
+        ├── __init__.py
+        └── solver.py
+```
+
+- The file `__init__.py` is required to import the `package_name/` as a package. This file is mostly empty
```

Comment:
```suggestion
- The file `__init__.py` is required to import the `package_name/` as a package. This file is normally empty
```
Maybe phrase it like this? I find "mostly" empty confusing. Is it empty or not? I personally have never put any content into a file called `__init__.py`. Are there cases where one would write something in there? If so, it would be nice to give an example (orally or on the slide) where the file has some content.